### ISSUE TITLE
Reference frame change breaks stabilisation

### ISSUE THREAD

**Issue by 0p3x** [mapoga/nuke-vector-matrix#37]
Hello, i've had some problem.
I have series of cornerpins, that stabilise computer monitor. I merge them together. Stabilisation is perfect. Then i try to change reference frame of a stailisation, by copying "to" into "from" parameters and removing animation. Stabilisation begins to wobble a bit.  Changing from 1000 to 1050 reference frame is a slight wobble. From 1000 to 1250 is a bigger one. Matrix is set to default

I've reproduced it into that script https://pastebin.com/XmUTyj3F

**Comment by herronelou** [comment]
Thanks for your message. I'll try to look into this as soon as possible. I might have some time this afternoon, but if not it may only be next week.
Thanks for your patience.

**Comment by herronelou** [comment]
It seems to be an order of operations problem, it's fairly easy to reproduce the results and also get the results that you expect:
https://pastebin.com/9ecFVDWi

Now I need to do some thinking, I'm not sure if I hooked things up backwards, or if there are use cases for both orders of operations.

**Comment by herronelou** [comment]
Hi @0p3x I have updated the code to use the different order of operations, I've tested a few other cases and found that this new order tends to give results that are closer to what I'd expect to see. 
Note that it means you can't properly change the reference frame on a CornerPin by affecting the From, you have to use the tool "Convert transforms" and set your reference frame there.
![image](https://github.com/mapoga/nuke-vector-matrix/assets/1182452/443841f6-414f-4388-8a4b-cb410101fd48)

**Comment by herronelou** [comment]
hummm, after more testing, both modes have pros and cons, I'll add an extra knob to choose.

**Comment by 0p3x** [comment]
Good catch on transformation orders :)
Thanks for a fast fix! Awesome tool, loving it a lot!

As i can see on a node info - its a container for several cornerpins. Is it possible to add aditional cornerpin to the end, with its transform, which is calling MergedPin's Label ? So it would be possible to change reference frame on a fly?

**Comment by herronelou** [comment]
I think you could do something like that, but it's probably a bit out of scope for what the tool currently does.

There are a few ways you can achieve this currently:

1. You can re-convert the node to a node of the same type with a different reference frame using the "convert transform"
2. You leave the initial node untouched, but changing its reference frame is equivalent to applying the inverse of the node at a specific frame, so you could duplicate your baked cornerpin, clear animation at frame 1001 (or any other frame you want to use as ref), invert it, and apply it after your main cornerpin.

**Comment by 0p3x** [comment]
Yeah, i was doing second option as a workaround, but adding more cornerpins is what i tried to avoid at the first place. And first option requires a lot of manual input. Usually i just tab-run a script and its automatically changes for a required frame. Fox example its usefull when you have one matchmove and a lof of frames to construct a cleanplate of. 
I think i will write a script of automated 1-button transform of original baked cornerpin rather than opening convert menu

Thanks!

**Comment by herronelou** [comment]
Yes that's a good idea if you do that a lot. Feel free to fork this repo and add something if you want to, all the backend functions should already exist here to do what you need, just needs to hook it up to a gui with less controls.
